src/input.rs

```diff
@@ -148,6 +148,10 @@ pub trait ExactSizeInput<'a>: Input<'a> {
 pub trait SliceInput<'a>: ExactSizeInput<'a> {
     /// The unsized slice type of this input. For [`&str`] it's `&str`, and for [`&[T]`] it will be `&[T]`.
     type Slice;
+    
+    /// Get the full slice of the input
+    #[doc(hidden)]
+    fn full_slice(&self) -> Self::Slice;
```

Comment:
Hmm, I'm not sure what this adds. What's the advantage of taking the full slice and using a sub-range rather than just using the required sub-slice?

Reply:
This is for the aforementioned fixes to anchors - the example of what this fixes is basically the canonical example on the documentation for `span`, but to explain more: imagine the regex `\b\w+\b` - this should match only a whole word, requiring a boundary at both sides. With a parser consuming a padded `foo`, this would be expected to match `foo bar` but not `foobar` - but currently, it will match both, because slicing the input down means the regex engine sees nothing before `bar`. With this change, the engine will start matching at `bar`, but be aware of whether ` ` or `foo` directly precedes it, fixing this behavior.

Reply:
Ah, nice. That explains all, thanks :)